dds/idl/typeobject_generator.cpp

```diff
@@ -509,3 +456,28 @@ typeobject_generator::gen_union(AST_Union* node, UTL_ScopedName* name,
   }
   return true;
 }
+
+void typeobject_generator::gen_type_flag_str(std::string& type_flag_str,
+                                             ExtensibilityKind exten,
+                                             bool can_be_mutable,
+                                             AST_Decl* node) {
+    switch (exten) {
+      case extensibilitykind_final:
+        type_flag_str = "XTypes::IS_FINAL";
+        break;
+      case extensibilitykind_appendable:
+        type_flag_str = "XTypes::IS_APPENDABLE";
+        break;
+      case extensibilitykind_mutable:
+        if (can_be_mutable) {
+          type_flag_str = "XTypes::IS_MUTABLE";
+        } else {
+          idl_global->err()->misc_error(
+          "Unexpected extensibility while setting flags: type cannot be mutable", node);
+        }
+        break;
+      default:
+        idl_global->err()->misc_error(
+          "Unexpected extensibility while setting flags", node);
+    }
+}
```

Comment:
Github says this is missing a final newline. Many text editors have an option to insert one automatically if they don't already do it by default.